packages/ra-core/src/routing/useGetPathForRecord.ts

```diff
@@ -52,33 +52,21 @@ export const useGetPathForRecord = <RecordType extends RaRecord = RaRecord>(
     const getPathForRecord = useGetPathForRecordCallback<RecordType>(options);
 
     // we initialize the path with the link value
-    const [path, setPath] = useState<string | false | undefined>(() => {
-        getPathForRecord({
-            record,
-            resource,
-            link,
-        }).then(resolvedLink => {
-            if (resolvedLink === false) {
-                // already set to false by default
-                return;
-            }
-            // update the path when the promise resolves
-            setPath(resolvedLink);
-        });
-
-        return false;
-    });
+    const [path, setPath] = useState<string | false | undefined>();
 
     // update the path if the record changes
     useEffect(() => {
-        getPathForRecord({
-            record,
-            resource,
-            link,
-        }).then(resolvedLink => {
+        const updatePath = async () => {
+            const resolvedLink = await getPathForRecord({
+                record,
+                resource,
+                link,
+            });
             // update the path when the promise resolves
             setPath(resolvedLink);
-        });
+        };
+
+        updatePath();
```

Comment:
My understanding is that you changed nothing here (except Promise => await). I don't think the result is better and it brings some noise in this PR, would you mind reverting this change?